src/c++/perf_analyzer/genai-perf/genai_perf/graphs/scatter_plot.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import pandas as pd
+import plotly.express as px
+from genai_perf.graphs.base_plot import BasePlot
+from genai_perf.llm_metrics import Statistics
+
+
+class ScatterPlot(BasePlot):
+    """
+    Generate a scatter plot in jpeg and html format.
+    """
+
+    def __init__(self, stats: Statistics) -> None:
+        super().__init__(stats)
+
+    def create_scatter_plot(
+        self,
+        x_key: str = "",
+        scale_x: bool = False,
+        y_key: str = "",
+        scale_y: bool = False,
+        graph_title: str = "",
+        x_label: str = "",
+        y_label: str = "",
+        filename_root: str = "",
+        preprocessed_x_data=[],
```

Comment:
You really don't want to have a mutable type (e.g. list) as a default value. This can lead to unexpected behavior. See more information [here](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html#:~:text=Passing%20mutable%20lists%20or%20dictionaries,that%20the%20function%20is%20called.).

Your current implementation should be fine since you do not update preprocessed_x_data and preprocessed_y_data (which is why CodeQL did not catch it), but moving to a different implementation such as using None would be better, since someone else could break it later. Also None should work fine with the "if not" statements.

Reply:
Cool. Learned something today. thanks for calling this out.